### ISSUE TITLE
Aborting ExternalLoginCallback causes error when trying to access a page which requires Authentication

### ISSUE THREAD

**Issue by madmunsterdaddy** [aspnet/Identity#860]
Steps to reproduce

1) Create a new web app with Individual User Accounts
2) Add Twitter authentication
3) Run the app
4) Click Log in
5) Click the Twitter button and enter twitter credentials
6) You're now at ExternalLoginCallback... don't add email address
7) Navigate to /Manage/Index

ArgumentNullException: Value cannot be null.
 Parameter name: user

**Comment by madmunsterdaddy** [comment]
You can get stuck in some nasty error loops if you get into this state. This should be handled more gracefully.

**Comment by pingcrosby** [comment]
I am having the same issues now  - how are you handling it?

**Comment by madmunsterdaddy** [comment]
I've not yet found anything I can do about it. I believe the issue may be in signinmanager. When you sign in with external you get that second cookie which its obviously unable to locate a user with that id in the system and that cookie is just lingering (I believe).

**Comment by HaoK** [comment]
The twitter cookie should only live for 5 minutes by default after which it will be expired.  That said, is the null ref coming from the View?

**Comment by madmunsterdaddy** [comment]
Yes, it happens when it tries to get the current user.

**Comment by pingcrosby** [comment]
Just as an aside - the same symptoms occur with Google and Microsoft providers.  If a user presses the browser back button for example at the ExternalLoginCallback page - the server hangs.

**Comment by HaoK** [comment]
This code lives in the ManageController, you can just add a null guard to prevent null users from calling UserManager APIs.  I'll file an issue in the Templates repo to add some more error checking.

**Comment by HaoK** [comment]
https://github.com/aspnet/Templates/issues/596

**Comment by madmunsterdaddy** [comment]
I think it's a bit deeper than that from what I can see. 

It tanks in UserManager.GetCurrentUserAsync().

In an instance where you use Guid for ID type you'll get this:

FormatException: Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).
System.Guid.TryParseGuidWithNoStyle(String guidString, GuidResult& result)
System.Guid.TryParseGuid(String g, GuidStyles flags, GuidResult& result)
System.Guid..ctor(String g)
System.ComponentModel.GuidConverter.ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, Object value)
Microsoft.AspNetCore.Identity.EntityFrameworkCore.UserStore`4.ConvertIdFromString(String id)
Microsoft.AspNetCore.Identity.EntityFrameworkCore.UserStore`4.FindByIdAsync(String userId, CancellationToken cancellationToken)
Microsoft.AspNetCore.Identity.UserManager`1.FindByIdAsync(String userId)
Microsoft.AspNetCore.Identity.UserManager`1.GetUserAsync(ClaimsPrincipal principal)
WebApplication1.Controllers.ManageController.GetCurrentUserAsync() in ManageController.cs
           return _userManager.GetUserAsync(HttpContext.User);
WebApplication1.Controllers.ManageController.<Index>d__7.MoveNext() in ManageController.cs
            var user = await GetCurrentUserAsync();
--- End of stack trace from previous location where exception was thrown ---

**Comment by HaoK** [comment]
So you have an empty string for your userId in your principal I'm guessing right now to cause this behavior?

**Comment by madmunsterdaddy** [comment]
No, it has my external login username.

**Comment by HaoK** [comment]
This might be due to https://github.com/aspnet/Identity/issues/848 where the external login cookie is getting smushed with normal application cookie.  Try turning off AutomaticAuthenticate for the non ApplicationCookies:
```
  AddIdentity(options => {
    options.Cookies.ExternalCookie.AutomaticAuthenticate = false;
 }
```

**Comment by madmunsterdaddy** [comment]
Ah-ha!

Yep, that results in the correct behavior where it just returns me to login screen.

**Comment by HaoK** [comment]
Cool this is already fixed for 1.0.0, so the workaround should only be temporary
